### PR TITLE
Downgrade translation-contracts

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because the v3 cant be installed anyway.

## Changelog

```markdown
### Fixed
- Remove support for translation-contracts until it can be installed without relying on dev dependencies
```

### CODE REVIEW

**Comment by VincentLanglet** [comment]
i'm not sure about the issue on ClassificationBundle.

Test / PHP 8.0 + highest + normal (pull_request) fail but 
Test / PHP 8.0 + highest + symfony/symfony:"5.3.*" (and 5.4 and 6.0) works.

Wouldn't it be better to update the dev-kit to only allow unstable dependency on the 6.0 build.
I think it would fix the issue.

**Comment by jordisala1991** [comment]
This fixed it in a better way, unless we can ensure the dependency is installable on builds here we shouldn't add support for new versions.

Yes, removing dev builds also could fix the problem, but the root issue is allowing a version that is not installable.

File: composer.json [hunk 1]
```diff
@@ -113,5 +105,13 @@
         "psr-4": {
             "Sonata\\AdminBundle\\Tests\\": "tests/"
         }
+    },
+    "config": {
+        "sort-packages": true
+    },
+    "extra": {
+        "branch-alias": {
```
Comment (core23) [inline composer.json]: You can remove the branch alias

File: composer.json [hunk 2]
```diff
@@ -93,17 +89,13 @@
         "symfony/yaml": "^4.4 || ^5.3 || ^6.0",
         "vimeo/psalm": "^4.9.2"
     },
+    "conflict": {
+        "doctrine/doctrine-bundle": ">=3",
+        "sonata-project/doctrine-orm-admin-bundle": "<=4.0.0-alpha-1"
```
Comment (core23) [inline composer.json]: IMHO we can remove this, because alpha releases could be buggy and we should not add conflicts for any buggy lib

**Review by core23 — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (jordisala1991) [reply composer.json]: Removed

File: composer.json (hunk 2 above)
Comment (jordisala1991) [reply composer.json]: Removed, I think the same.

**Review by core23 — APPROVED** [review]

**Review by VincentLanglet — CHANGES_REQUESTED** [review]
> Yes, removing dev builds also could fix the problem, but the root issue is allowing a version that is not installable.

The fact the build works fine when we explicit set symfony to 4.4, 5.3, 5.4 or 6.0 means to me that this library is compatible.
The issue is more with the fact that on the `highest` strategy, composer is mixing different symfony version and ends up with incompatible one.

Look at the build
https://github.com/sonata-project/SonataClassificationBundle/runs/4422119398?check_suite_focus=true

The issue is that we're trying to use a RC version instead of the right one.
`Locking symfony/string (v5.4.0-RC1)`

To me it's not an admin bundle issue and will be fixed with the dev-kit PR

**Comment by jordisala1991** [comment]
Still, the problem it is allowing versions that are not installable, and this fixes it. We should not allow versions that cant be installed because we are not sure if they work. We are doing like this for quite a lot of time, not sure why you want to change that now, but imo thats what I am trying to fix. Unless you can prove version 3 is installable we should remove it.

**Comment by VincentLanglet** [comment]
> Still, the problem it is allowing versions that are not installable, and this fixes it. We should not allow versions that cant be installed because we are not sure if they work. We are doing like this for quite a lot of time, not sure why you want to change that now, but imo thats what I am trying to fix. Unless you can prove version 3 is installable we should remove it.

I can install it
image

**Comment by jordisala1991** [comment]
Take a look at what happens with tests if you do that. It is downgrading symfony extra to a version that it can conflict with symfony string (Problem shown at classification bundle)

**Comment by VincentLanglet** [comment]
> Take a look at what happens with tests if you do that. It is downgrading symfony extra to a version that it can conflict with symfony string (Problem shown at classification bundle)

Indeed because the conflict was only added in 5.4 https://github.com/symfony/string/commit/9ffaaba53c61ba75a3c7a3a779051d1e9ec4fd2d

**Review by VincentLanglet — APPROVED** [review]

**Pull request merged**